sdk/iot/hub/src/az_iot_hub_client_twin.c

```diff
@@ -11,6 +11,7 @@
 
 #include <_az_cfg.h>
 
+static const uint8_t null_terminator = '\0';
```

Comment:
Nit: I prefer using the actual chars throughout the code instead of defining them here. This makes the code harder to read as people now need to go to the top of the file to understand what these are.
I think it's useful to declare constants when they might change in the future or their meaning is unclear. The null terminator in C is always '\0'. Same goes for the characters below.

Reply:
I'm willing to go one way or another. Is it okay if we have it open for a team discussion and go whichever way we collectively decide? Democracy and all